src/library/scala/math/BigDecimal.scala

```diff
@@ -122,7 +122,18 @@ object BigDecimal {
   /** Constructs a `BigDecimal` that exactly represents the number
    *  specified in a `String`.
    */
-  def exact(s: String): BigDecimal = exact(new BigDec(s))
+  def exact(s: String): BigDecimal =
+    try exact(new BigDec(s))
+    catch {
+      // if it fails because exp is MinValue, construct by hand.
+      // BigDec throws for reasons of backward compatibility.
+      case e: NumberFormatException if s endsWith "+2147483648" =>
+        val r = """(\+|-)?(\d+)(\.\d+)?(e|E)\+2147483648""".r
```

Comment:
The `+` might be expected, but it's not actually necessary for the BigDecimal constructor, so you should just take all the NumberFormatExceptions and match on r (and make `+` optional).  This has cleaner logic also: you don't have to worry about long strings of numbers in two spots.